src/main/java/org/jenkinsci/plugins/casc/Configurator.java

```diff
@@ -231,14 +232,15 @@ private Klass getKlass() {
      *      a Jenkins object is configured.
      * @return
      *      Fully configured Jenkins object that results from this configuration.
+     * @throws Exception if something went wrong, depends on the concrete implementation
```

Comment:
FTR I started refactoring that and never finished so far. Usage of `throws Exception` in APIs is a pretty bad design pattern in Java